alacritty/src/event.rs

```diff
@@ -816,36 +818,49 @@ impl<'a, N: Notify + 'a, T: EventListener> input::ActionContext<T> for ActionCon
         }
     }
 
-    /// Paste a text into the terminal.
-    fn paste(&mut self, text: &str, bracketed: bool) {
+    /// Write text into the terminal or widget.
+    fn write_str(&mut self, text: &str, mode: WriteStrMode) {
         if self.search_active() {
             for c in text.chars() {
                 self.search_input(c);
             }
-        } else if bracketed && self.terminal().mode().contains(TermMode::BRACKETED_PASTE) {
-            self.on_terminal_input_start();
 
-            self.write_to_pty(&b"\x1b[200~"[..]);
+            return;
+        }
 
-            // Write filtered escape sequences.
-            //
-            // We remove `\x1b` to ensure it's impossible for the pasted text to write the bracketed
-            // paste end escape `\x1b[201~` and `\x03` since some shells incorrectly terminate
-            // bracketed paste when they receive it.
-            let filtered = text.replace(['\x1b', '\x03'], "");
-            self.write_to_pty(filtered.into_bytes());
+        self.on_terminal_input_start();
 
-            self.write_to_pty(&b"\x1b[201~"[..]);
-        } else {
-            self.on_terminal_input_start();
-
-            // In non-bracketed (ie: normal) mode, terminal applications cannot distinguish
-            // pasted data from keystrokes.
-            // In theory, we should construct the keystrokes needed to produce the data we are
-            // pasting... since that's neither practical nor sensible (and probably an impossible
-            // task to solve in a general way), we'll just replace line breaks (windows and unix
-            // style) with a single carriage return (\r, which is what the Enter key produces).
-            self.write_to_pty(text.replace("\r\n", "\r").replace('\n', "\r").into_bytes());
+        match mode {
+            WriteStrMode::BracketedOrMultiline => {
```

Comment:
Hm if we just make this a boolean and call it `raw` instead of bracketed, then it's not confusing but we also don't need an enum. Would that be easier maybe?